### DISCUSSION TITLE
Isolating an orientation on a pole figure

### DISCUSSION THREAD

**Discussion by RellieGoddard** [mtex-toolbox/mtex#1984] [Ask Anything] [answered]
Hello! 

I'm currently working on app in Matlab, which at one point looks to allow the user to select orientations from a pole figure and see which grains have that orientation. So far I've managed to isolate X/Y co-ordinates using the the 'clickTracker' tool (see below and Example_1). 

```
figure(app.UIFigure)
            app.clickTracker = false;
            while ~app.clickTracker
                pause(0.001)
            end
            pick_x = app.clickData(1);
            pick_y = app.clickData(2);
            hold(app.UIAxes, 'on');
            a = scatter(app.UIAxes, pick_x, pick_y, 's', 'filled', 'MarkerFaceColor','r')
            title(app.UIAxes,['x = ', num2str(pick_x), ' y =', num2str(pick_y)]);

            app.x_points(end+1) = [pick_x];
            app.y_points(end+1) = [pick_y];

            app.point_handles(end+1) = a;
```

![Example_1](https://github.com/mtex-toolbox/mtex/assets/52534240/42ba7d3a-4d06-41c9-9d89-ffa6f06c43ae)

However, I've found the X/Y co-ordinate that I pick are not in the same reference frame as the X/Y co-ordinates for each point in the ODF (found through multiplying the orientation by the crystal symmetry). When not in the app (Example_2), I know you can click on a pole figure and it gives you the information associated with that point, but I'm not sure what these numbers actually related to (e.g., phi1/Phi/phi2).  Also, this action is not possible once the pole figure is plot in the app axes.

![Example_2](https://github.com/mtex-toolbox/mtex/assets/52534240/870bbbad-0efd-4479-8ccb-b8ae67339edd)

Any ideas about how to go about this would be much appreciated! 

Thanks, 
Rellie

**Comment by kilir** [comment, marked as answer]
Hi Rellie,
with respect to the info you will get from the cursor within an mtex figure: If you plot discrete orientations, it will give you the id of the point, polar coordinates and actually mtex keeps track of the points plotted, hence is able to tell from which orientation the point war derived.

![Screenshot_20231019_175804](https://github.com/mtex-toolbox/mtex/assets/11697557/7becbb1a-4fbc-4d9f-af88-27d51bbcc004)

For a density plot, you will of course only get the polar coordinates as well as the value.

![Screenshot_20231019_175938](https://github.com/mtex-toolbox/mtex/assets/11697557/0ec18667-9269-4b22-a87a-2ca4f09ff2d8)


With respect to your app: a) what you probably read out are the x,y coordiantes within the image plane which do not honor the spherical projection not the plotting area etc. Reversing this is most likely not a huge fun. b) Getting a crystallographic orientation from a single pole figure is not possible at all since it is not sufficiently defined.

I hope this helps.

Cheers,
Rüdiger

**Comment by RellieGoddard** [comment]
Thanks Rüdiger! It seems like what I originally had in mind isn't going to be possible, so I'll have to rethink. 

All the best, 
Rellie
